src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Snackbar/SnackBar.android.cs

```diff
@@ -70,6 +77,21 @@ internal void Show(Page sender, SnackBarOptions arguments)
 			snackBar.Show();
 		}
 
+		static async Task<IVisualElementRenderer> GetRendererWithRetries(Page sender, int maxRetries = 5)
+		{
+			var retriesLeft = maxRetries;
+
+			var renderer = Platform.GetRenderer(sender);
+			while (renderer == null && retriesLeft > 0)
+			{
+				retriesLeft--;
+				await Task.Delay(50);
```

Comment:
still not an ideal solution.
it would be better to investigate why the renderer is null (is it XF issue or you call DisplayXXX too early - before the page is initialized).

Reply:
It's a XF Shell bug. We cannot control when we show SnackBar since we show it when user asks us to. And issue happens when we show SnackBar on one page and immediately switch to the different page. User can switch pages from the UI, so we have no way of knowing if we currently transitioning between pages (well, we can check renderer for `null`).

Reply:
I agree with @VladislavAntonyuk at this, we should move with another approach, I liked the way how Shane proposed, listening the `PropertyChanged` and using a `TaskCompletionSource`, we also could add a `CancelationToken` with a timeout to avoid dead-locks.

Little snippet provided by Shane on our discord channel.
```csharp
if(Platform.GetrendereR(page) == null)
 page.ElementPropertyFChanged += WatchForRendererToSet 


void WatchForRendererToSet (_, args) 
if(args.PropertyName == "Renderer") 
{
      page.ElementPropertyChanged -= WatchForRendererToTest
       task.SetResult(Platform.GetRenderer(page));
}

```

Reply:
@pictos I thought this PR will never be merged anyway (with any implementation), since the bug is in XF, and not in this lib.

I don't agree with this. If this lib provides some functionality, it should ensure that is works (even if that mean including workarounds for bugs in 3rd party code). But that's not my decision to make.

So will this PR be able to move forward if I do the implementation you described?

Reply:
@maxkoshevoi, I agree in parts with you. I agree that our lib breaking an app is really bad, even if the issue isn't here. Also, we need to try to minimize that impact by finding a workaround for it. And I believe (here is where I disagree with you) that we shouldn't add a lot of code to work around a bug, mainly if this workaround changes various APIs.

My suggestion also adds a lot of changes to this API, so maybe isn't the better of the world.
The short answer is don't work on this right now, I'll discuss with other members and see which approach we will do.
1. Asking for devs to use the workaround;
2. make our API work with this bug.

Reply:
I agree that we shouldn't add a lot of code just to workaround the bug, but this workaround is not that big. It just adds one method:

![image](https://user-images.githubusercontent.com/6609929/114324001-865e6200-9b30-11eb-93d4-c64fae4f8f65.png)

The rest is just about making `StanckBar.Show` method return `Task` instead of `void`
![image](https://user-images.githubusercontent.com/6609929/114324036-b6a60080-9b30-11eb-9bfd-1cec42cdef93.png)

So there is no logic change there, I basically just added `await`s where this method is called

Reply:
We also make some methods `async`, that we discussed to not do in this [PR](https://github.com/xamarin/XamarinCommunityToolkit/pull/973)

Reply:
@pictos This satiation is a bit different. Currently we have one task that we return from our public `ShowSnackBar` method.

Since nothing in this method depends on that task, there are two ways to pass it to the caller:
- If `ShowSnackBar` is not `async`, `Task` can be returned from the method (current implementation)
- If `ShowSnackBar` is `saync`, `Task` can be `await`ed (chnage proposed in that PR)

Result is pretty much the same.

But is this PR we have two tasks, since `snackBar.Show` is now also returns a `Task`. So now we need to make `ShowSnackBar` `async` in order to `await` `snackBar.Show` and preserve execution flow.

Reply:
@pictos Interestingly, this doesn't fix the issue. 
```cs
static Task<IVisualElementRenderer> GetRenderer(Page page)
{
	TaskCompletionSource<IVisualElementRenderer> tcs = new ();

	var renderer = Platform.GetRenderer(page);
	if (renderer != null)
	{
		tcs.SetResult(renderer);
	}
	else
	{
		page.PropertyChanged += WatchForRendererToSet;
		void WatchForRendererToSet(object sender, PropertyChangedEventArgs args)
		{
			if (args.PropertyName == "Renderer")
			{
				page.PropertyChanged -= WatchForRendererToSet;
				tcs.SetResult(Platform.GetRenderer(page));
			}
		}
	}

	return tcs.Task;
}
```

Method itself works as expected, but SnackBar creation fails based on a renderer that is returned here `tcs.SetResult(Platform.GetRenderer(page));`

![image](https://user-images.githubusercontent.com/6609929/114719639-f68f0280-9d3f-11eb-8d4f-22ecc3e495a5.png)

Anyway, fix using retries works, so leaving it as it.